component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPKeyGenerator.java

```diff
@@ -132,6 +133,7 @@
 
             String issuer = TOTPUtil.getTOTPIssuerDisplayName(tenantDomain, context);
             String displayUsername = TOTPUtil.getTOTPDisplayUsername(tenantAwareUsername);
+            displayUsername = getTOTPIssuerDisplayNameForFederatedUser(context, displayUsername);
```

Comment:
Check whether this is a federated user and call this. Otherwise this is confusing for local users.